src/Controller/UserController.php

```diff
@@ -80,50 +81,55 @@ public function getUserDetails(User $user): Response
      * @throws ORMException
      * @throws OptimisticLockException
      */
-    public function editUser($id, Request $request, UserRepository $userRepository)
+    public function editUser($id, Request $request, UserRepository $userRepository, UserInterface $user)
     {
-        $user = $userRepository->find($id);
-        if (!$user) {
-            return new JsonResponse(['message' => 'The user was not found!'], Response::HTTP_NOT_FOUND);
+        $userUsername = $user->getUsername();
+        $userPage = $userRepository->find($id);
+        if ($userPage === null) {
+            return new JsonResponse(['message' => 'Access denied!'], Response::HTTP_NOT_FOUND);
         }
 
-        $data = $request->getContent();
+        $userPage = $userRepository->find($id)->getUsername();
+        if ($userUsername === $userPage) {
```

Comment:
Compare user ids, not usernames ..

Reply:
It should be the reverse. If the ids (or usernames in your case) don't match, return the error JSON, as so, you won't have too much identation and you should alway fail / return fast from the functions.